channels/channel.nos/nos2010/chn_nos2010.py

```diff
@@ -1331,6 +1332,81 @@ def update_video_item_live(self, item):
         item.complete = True
         return item
 
+    def create_iptv_streams(self, parameter_parser):
+        """ Fetch the available live channels using EPG endpoint and format them into JSON-STREAMS
+
+        :param ActionParser parameter_parser: a ActionParser object to is used to parse and
+                                                   create urls
+
+        :return: Formatted stations
+        :rtype: list
+        """
+
+        epg_data = JsonHelper.loads(UriHandler.open(datetime.datetime.now().strftime("https://start-api.npo.nl/epg/%Y-%m-%d?type=tv"),
+            no_cache=True,
+            additional_headers=self.__jsonApiKeyHeader))
+
+        parent_item = MediaItem("Live", "https://www.npostart.nl/live", media_type=mediatype.FOLDER)
+        items = []
+        iptv_streams = []
+
+        for stations in epg_data["epg"]:
+            livestream = stations["channel"]["liveStream"]
+            item = MediaItem(livestream["title"], livestream["shareUrl"], media_type=mediatype.VIDEO)
+            items.append(item)
+
+            iptv_streams.append(dict(
+                id=livestream["id"],
+                name=livestream["title"],
+                logo=livestream["images"]["original"]["formats"]["tv"]["source"],
+                group=self.channelName,
+                stream=parameter_parser.create_action_url(self, action=action.PLAY_VIDEO, item=item, store_id=parent_item.guid),
+            ))
+        
+        parameter_parser.pickler.store_media_items(parent_item.guid, parent_item, items)
```

Comment:
Keep in mind, that the pickle storage is cleared on a regular basis. So it might fail after some time. Is that an issue?

Reply:
Oh, then there is not issue! In `pickler.py` there is a `purge_store` that is set for 30 days. So older pickles will be deleted.